bench/src/bulk.rs

```diff
@@ -80,34 +136,143 @@ async fn client(server_addr: SocketAddr, server_cert: quinn::Certificate) -> Res
     client_config
         .add_certificate_authority(server_cert)
         .unwrap();
+    let mut client_config = client_config.build();
+    client_config.transport = Arc::new(transport_config());
+
     let quinn::NewConnection { connection, .. } = endpoint
-        .connect_with(client_config.build(), &server_addr, "localhost")
+        .connect_with(client_config, &server_addr, "localhost")
         .unwrap()
         .await
         .context("unable to connect")?;
     trace!("connected");
 
-    let mut stream = connection
-        .open_uni()
-        .await
-        .context("failed to open stream")?;
-    const DATA: &[u8] = &[0xAB; 1024 * 1024];
     let start = Instant::now();
-    for _ in 0..1024 {
-        stream
-            .write_all(DATA)
-            .await
-            .context("failed sending data")?;
+
+    let connection = Arc::new(connection);
+    let get_connection = || connection.clone();
+
+    let mut ops = futures::stream::iter((0..config.streams).map(|_| async move {
+        send_data_on_stream(get_connection(), config.stream_size_mb).await
+    }))
+    .buffer_unordered(config.max_streams);
+
+    let mut total_size = 0;
+    let mut duration_hist: Histogram<u64> = Histogram::new(3).unwrap();
```

Comment:
Can we remove the need for the type annotations here by adding a `u64` suffix to the `new()` parameter?

Reply:
done